lib/transform.js

```diff
@@ -87,235 +42,34 @@ function transform(source, template, options = {}) {
-    }
-  });
-
-  let attrs = [];
-  if (elementId) {
-    attrs.push(b.attr('id', b.text(elementId)));
-  }
-  attributeBindings.forEach((value, key) => {
-    attrs.push(b.attr(key, b.mustache(`this.${value}`)));
-  });
-  if (classNodes.length === 1) {
-    attrs.push(b.attr('class', classNodes[0]));
-  } else if (classNodes.length !== 0) {
-    let parts = [];
-    classNodes.forEach((node, i) => {
-      if (i !== 0) parts.push(b.text(' '));
-      parts.push(node);
-    });
-
-    attrs.push(b.attr('class', b.concat(parts)));
-  }
-  attrs.push(b.attr('...attributes', b.text('')));
-
-  templateAST.body = [
-    b.element(tagName, {
-      attrs,
-      children: [b.text(`\n${PLACEHOLDER}\n`)],
-    }),
-  ];
-
-  let newTemplate = templateRecast.print(templateAST).replace(PLACEHOLDER, indentLines(template));
-
-  return { source: newSource, template: newTemplate, tagName };
-}
-
-function isProperty(path, name) {
-  let node = path.value;
-  return node.type === 'ObjectProperty' && node.key.type === 'Identifier' && node.key.name === name;
-}
-
-function isMethod(path, name) {
-  let node = path.value;
-  return node.type === 'ObjectMethod' && node.key.type === 'Identifier' && node.key.name === name;
-}
-
-function findStringProperty(properties, name, defaultValue = null) {
-  let propertyPath = properties.filter(path => isProperty(path, name))[0];
-  if (!propertyPath) {
-    return defaultValue;
+  if (exportDefaultDeclarations.length === 1) {
+    return 'classic';
   }
-
-  let valuePath = propertyPath.get('value');
-  if (valuePath.value.type !== 'StringLiteral') {
-    throw new SilentError(`Unexpected \`${name}\` value: ${j(valuePath).toSource()}`);
-  }
-
-  return valuePath.value.value;
 }
 
-function findTagName(properties) {
-  return findStringProperty(properties, 'tagName', 'div');
-}
-
-function findElementId(properties) {
-  return findStringProperty(properties, 'elementId');
-}
-
-function findStringArrayProperties(properties, name) {
-  let propertyPath = properties.filter(path => isProperty(path, name))[0];
-  if (!propertyPath) {
-    return [];
-  }
-
-  let arrayPath = propertyPath.get('value');
-  if (arrayPath.value.type !== 'ArrayExpression') {
-    throw new SilentError(`Unexpected \`${name}\` value: ${j(arrayPath).toSource()}`);
-  }
-
-  return arrayPath.get('elements').value.map(element => {
-    if (element.type !== 'StringLiteral') {
-      throw new SilentError(`Unexpected \`${name}\` value: ${j(arrayPath).toSource()}`);
-    }
-
-    return element.value;
-  });
-}
+function transform(source, template, options = {}) {
+  let root = j(source);
+  let type = checkComponentType(root);
+  let result;
 
-function findAttributeBindings(properties) {
-  let attrBindings = new Map();
-  for (let binding of findStringArrayProperties(properties, 'attributeBindings')) {
-    let [value, attr] = binding.split(':');
-    attrBindings.set(attr || value, value);
+  switch (type) {
```

Comment:
can you change this to be an if-condition? I'm slightly allergic to `switch`... 😅

Reply:
Sure, done.